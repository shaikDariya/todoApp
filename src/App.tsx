import { ToastContainer } from "react-toastify";
import { CssBaseline, Container } from "@mui/material";
import { ThemeProvider, createTheme } from "@mui/material/styles";

import Todo from "@/pages/Todo";
import { TodoProvider } from "@/context/todo";

import "react-toastify/dist/ReactToastify.css";
import "./app.css";

const theme = createTheme({
  palette: {
    primary: {
      main: "#4caf50",
    },
    secondary: {
      main: "#e8eaed",
    },
  },
});

function App() {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Container maxWidth="xl" sx={{ bgcolor: "secondary.main" }}>
        <TodoProvider>
          <Todo />
        </TodoProvider>
      </Container>
      <ToastContainer />
    </ThemeProvider>
  );
}

export default App;
