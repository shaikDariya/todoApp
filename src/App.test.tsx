import { render, screen } from "@testing-library/react";
import App from "./App";

test("it should have the correct title", () => {
  render(<App />);
  const title = screen.getByText("Todo List");
  expect(title).toBeInTheDocument();
});

test("it should match snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
