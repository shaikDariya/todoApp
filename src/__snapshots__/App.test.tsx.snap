// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-121xoxs-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-4gqh1i-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-escpob-MuiTypography-root"
      >
        Todo List
      </h4>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-1wismux-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-gvxdmm-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-150d6dw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              Add New Todo
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1m3tev7-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Add New Todo
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-ke5b6m-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                />
              </svg>
            </span>
            Add
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-wcb8xa-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ihiw17-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ihiw17-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ihiw17-MuiSkeleton-root"
        />
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
